src/schemas.jl

```diff
@@ -856,12 +856,19 @@ end
 ##### Base overload definitions
 #####
 
-function Base.:(==)(x::T, y::T) where {T<:AbstractRecord}
-    return all(i -> getfield(x, i) == getfield(y, i), 1:fieldcount(T))
+_typeof(r::AbstractRecord) = Base.unwrap_unionall(Base.typename(typeof(r)).wrapper)
```

Comment:
What if we generated methods for `_typeof` in the macro? Could avoid internals while still having fewer methods overall